soc/arm/ambiq/apollo4x/soc.c

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright (c) 2023 Antmicro <www.antmicro.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "hal/am_hal_utils.h"
+#include <zephyr/arch/cpu.h>
+#include <zephyr/init.h>
+#include <zephyr/kernel.h>
+#include <zephyr/logging/log.h>
+
+#include <am_mcu_apollo.h>
+#include <am_bsp.h>
+
+#include <soc.h>
+
+LOG_MODULE_REGISTER(soc, CONFIG_SOC_LOG_LEVEL);
+
+am_hal_gpio_pincfg_t pin_rx_config = {
+	.GP.cfg_b.uFuncSel             = AM_HAL_PIN_47_UART0RX,
+	.GP.cfg_b.eGPInput             = AM_HAL_GPIO_PIN_INPUT_NONE,
+	.GP.cfg_b.eGPRdZero            = AM_HAL_GPIO_PIN_RDZERO_READPIN,
+	.GP.cfg_b.eIntDir              = AM_HAL_GPIO_PIN_INTDIR_NONE,
+	.GP.cfg_b.eGPOutCfg            = AM_HAL_GPIO_PIN_OUTCFG_DISABLE,
+	.GP.cfg_b.eDriveStrength       = AM_HAL_GPIO_PIN_DRIVESTRENGTH_0P1X,
+	.GP.cfg_b.uSlewRate            = 0,
+	.GP.cfg_b.ePullup              = AM_HAL_GPIO_PIN_PULLUP_NONE,
+	.GP.cfg_b.uNCE                 = 0,
+	.GP.cfg_b.eCEpol               = AM_HAL_GPIO_PIN_CEPOL_ACTIVELOW,
+	.GP.cfg_b.uRsvd_0              = 0,
+	.GP.cfg_b.ePowerSw             = AM_HAL_GPIO_PIN_POWERSW_NONE,
+	.GP.cfg_b.eForceInputEn        = AM_HAL_GPIO_PIN_FORCEEN_NONE,
+	.GP.cfg_b.eForceOutputEn       = AM_HAL_GPIO_PIN_FORCEEN_NONE,
+	.GP.cfg_b.uRsvd_1              = 0,
+};
```

Comment:
please implement a proper pinctrl driver

Reply:
Thanks for a quick response, will create a proper pinctrl driver.

Reply:
@gmarull: I have updated the PR to include a pinctrl driver